pasqal_cloud/client.py

```diff
@@ -230,18 +206,22 @@ def get_batch_jobs(self, batch_id: str) -> list[Dict[str, Any]]:
             # the same order as they do in the GET /batches/{id} response
         )
 
+    @retry_http_error(max_retries=2)
+    def _download_results(self, results_link: str) -> JobResult:
+        response = requests.get(results_link)
```

Comment:
As much as I don't care what interface you're using for your HTTP request. We are now splitting our use between the internal `_request(method_type: str)` and `requests.method_type()` so maybe we should pick one or the other for consistency

Reply:
From my understanding `_request` is used to speak to our APIs and automatically attaches authentication headers to the request; we don't want these when fetching the results in s3, which is why we use `request` directly here.

We should maybe clarify this by renaming the `_request` method to `_authenticated_request` or something. We could also add in the docstring that this method is meant to be used when calling our APIs.

Or we could make the `_request` function more generic to make auth headers optional.

I have a preference for the first option personally.

Reply:
Do we want to rename it in this PR? If so, I will do it as the last commit before merging the PR to facilitate review

Reply:
Yeah since this MR introduces another method to make http request, I think it should take care of renaming to keep things clear